src/main/java/com/enonic/lib/graphql/scalars/DateTimeScalar.java

```diff
@@ -0,0 +1,126 @@
+                    if ( !( input instanceof String ) )
+                    {
+                        throw new CoercingSerializeException(
+                            String.format( "Expected something what can convert to 'java.time.temporal.Temporal' but was '%s'",
+                                           Kit.typeName( input ) ) );
+                    }
+
+                    try
+                    {
+                        LocalDateTime ldt = LocalDateTime.parse( input.toString(), DateTimeFormatter.ISO_LOCAL_DATE_TIME );
+                        return DateTimeFormatter.ISO_LOCAL_DATE_TIME.format( ldt );
+                    }
+                    catch ( DateTimeParseException e )
+                    {
+                        OffsetDateTime odt = OffsetDateTime.parse( input.toString(), DateTimeFormatter.ISO_OFFSET_DATE_TIME );
+                        return DateTimeFormatter.ISO_OFFSET_DATE_TIME.format( odt );
+                    }
+                }
+            }
+            catch ( DateTimeException e )
+            {
+                throw new CoercingSerializeException( e );
+            }
+        }
+
+        @Override
+        public Temporal parseValue( Object input )
+            throws CoercingParseValueException
+        {
+            if ( input instanceof Temporal )
+            {
+                return (Temporal) input;
+            }
+
+            if ( !( input instanceof String ) )
+            {
+                throw new CoercingParseValueException( String.format( "Expected a 'String' but was '%s'.", Kit.typeName( input ) ) );
+            }
+
+            return parseInputToTemporal( input.toString() );
+        }
+
+        @Override
+        public Temporal parseLiteral( Object input )
+            throws CoercingParseLiteralException
+        {
+            if ( !( input instanceof StringValue ) )
+            {
+                throw new CoercingParseLiteralException( String.format( "Expected 'StringValue' but was '%s'.", Kit.typeName( input ) ) );
+            }
+
+            final String inputValue = ( (StringValue) input ).getValue();
+            return parseInputToTemporal( inputValue );
+        }
+
+        private Temporal parseInputToTemporal( final String inputAsString )
+        {
+            try
+            {
+                return LocalDateTime.parse( inputAsString, DateTimeFormatter.ISO_LOCAL_DATE_TIME );
+            }
+            catch ( DateTimeParseException e )
+            {
+                try
+                {
+                    return OffsetDateTime.parse( inputAsString, DateTimeFormatter.ISO_OFFSET_DATE_TIME );
+                }
+                catch ( DateTimeParseException ex )
+                {
+                    throw new CoercingSerializeException( ex );
+                }
+            }
+        }
+    };
+
+    public static GraphQLScalarType newDateTime()
+    {
+        return GraphQLScalarType.newScalar().
+            name( "CustomDateTime" ).
```

Comment:
We could think of nicer name

Reply:
Any ideas? We can not use DateTime, because this type name occupied by ExtendedScalars

Reply:
Renamed to LocalDateTime and LocalTime